src/encoded/tests/test_search.py

```diff
@@ -642,3 +642,227 @@ def test_barplot_aggregation_endpoint(workbook, testapp):
+        expected = self.DEFAULT_FACETS + [facet]
+        actual = [facet['field'] for facet in facets]
+        assert sorted(expected) == sorted(actual)
+
+    def test_search_multiple_additional_facets(self, testapp, hidden_facet_test_data):
+        """ Tests that enabling multiple additional facets works """
+        facets = testapp.get('/search/?type=TestingHiddenFacets'
+                             '&additional_facet=unfaceted_string'
+                             '&additional_facet=unfaceted_integer').json['facets']
+        expected = self.DEFAULT_FACETS + self.ADDITIONAL_FACETS
+        for facet in facets:
+            assert facet['field'] in expected
+            if facet['field'] == 'unfaceted_integer':
+                assert facet['aggregation_type'] == 'stats'
+            else:
+                assert facet['aggregation_type'] == 'terms'
+
+    @pytest.mark.parametrize('facet', DEFAULT_HIDDEN_FACETS)
+    def test_search_one_additional_default_hidden_facet(self, testapp, hidden_facet_test_data, facet):
+        """ Tests that passing default_hidden facets to additional_facets works correctly """
+        facets = testapp.get('/search/?type=TestingHiddenFacets&additional_facet=%s' % facet).json['facets']
+        expected = self.DEFAULT_FACETS + [facet]
+        actual = [facet['field'] for facet in facets]
+        assert sorted(expected) == sorted(actual)
+
+    def test_search_multiple_additional_default_hidden_facets(self, testapp, hidden_facet_test_data):
+        """ Tests that passing multiple hidden_facets as additionals works correctly """
+        facets = testapp.get('/search/?type=TestingHiddenFacets'
+                             '&additional_facet=last_name'
+                             '&additional_facet=sid').json['facets']
+        expected = self.DEFAULT_FACETS + self.DEFAULT_HIDDEN_FACETS
+        for facet in facets:
+            assert facet['field'] in expected
+            if facet['field'] == 'sid':
+                assert facet['aggregation_type'] == 'stats'
+            else:
+                assert facet['aggregation_type'] == 'terms'
+
+    @pytest.mark.parametrize('_facets', [
+        ['last_name', 'unfaceted_integer'],  # second slot holds number field
+        ['unfaceted_string', 'sid']
+    ])
+    def test_search_mixing_additional_and_default_hidden(self, testapp, hidden_facet_test_data, _facets):
+        """ Tests that we can mix additional_facets with those both on and off schema """
+        facets = testapp.get('/search/?type=TestingHiddenFacets'
+                             '&additional_facet=%s'
+                             '&additional_facet=%s' % (_facets[0], _facets[1])).json['facets']
+        expected = self.DEFAULT_FACETS + _facets
+        actual = [facet['field'] for facet in facets]
+        assert sorted(expected) == sorted(actual)
+        for facet in facets:
+            if facet['field'] == _facets[1]:  # second slot holds number field
```

Comment:
If `_facets[1]` were `sample_number_field` or some such, then it wouldn't need this comment.